src/core/qgspointlocator.h

```diff
@@ -309,6 +339,31 @@ class CORE_EXPORT QgsPointLocator : public QObject
      */
     MatchList verticesInRect( const QgsPointXY &point, double tolerance, QgsPointLocator::MatchFilter *filter = nullptr, bool relaxed = false );
 
+    /**
+     * Find centroids within a specified recangle
+     * Optional filter may discard unwanted matches.
+     * \since QGIS 3.12
+     */
+    MatchList centroidsInRect( const QgsRectangle &rect, QgsPointLocator::MatchFilter *filter = nullptr );
+
+    /**
+     * Override of centroidsInRect that construct rectangle from a center point and tolerance
+     * \since QGIS 3.12
+     */
+    MatchList centroidsInRect( const QgsPointXY &point, double tolerance, QgsPointLocator::MatchFilter *filter = nullptr );
+
+    /**
+     * Find middle segment within a specified recangle
+     * Optional filter may discard unwanted matches.
+     * \since QGIS 3.12
+     */
+    MatchList middlesInRect( const QgsRectangle &rect, QgsPointLocator::MatchFilter *filter = nullptr );
+
+    /**
+     * Override of middleInRect that construct rectangle from a center point and tolerance
+     * \since QGIS 3.12
+     */
+    MatchList middlesInRect( const QgsPointXY &point, double tolerance, QgsPointLocator::MatchFilter *filter = nullptr );
```

Comment:
These functions do not seem to be used anywhere and not tested either. I am wondering whether we need them at all? Maybe better to remove them if they are unused and untested?

Reply:
Yes, I had anticipated, but it's not useful at the moment, so better to remove then.